sklego/meta/grouped_predictor.py

```diff
@@ -212,112 +122,32 @@ def fit(self, X, y=None):
         self : GroupedPredictor
             The fitted estimator.
         """
-        X_group, X_value = _split_groups_and_values(
-            X, self.groups, min_value_cols=0, check_X=self.check_X, **self._check_kwargs
-        )
-
-        X_group = self.__add_shrinkage_column(X_group)
 
-        if y is not None:
-            y = check_array(y, ensure_2d=False)
+        # TODO: Validate class params?
 
-        if self.shrinkage is not None:
-            self.__set_shrinkage_function()
+        if is_classifier(self.estimator):
+            self.classes_ = np.sort(np.unique(y))  # TODO: Must be sequential for the rest of the code to work
+            self.n_classes_ = len(self.classes_)
 
-        # List of all hierarchical subsets of columns
-        self.group_colnames_hierarchical_ = expanding_list(X_group.columns, list)
+        self.groups_ = as_list(self.groups)
 
-        self.fallback_ = None
+        # TODO: __grouped_predictor_target_value__?
+        frame = pd.DataFrame(X).assign(__target_value__=np.array(y)).reset_index(drop=True)
```

Comment:
`__target_value__` (and `__global_model__` just right after) could be "safe enough" column names, but we can even prefix with `__grouped_predictor` or `__sklego_grouped_predictor`  to be extra safe that someone is not using those as column names 😂